File: compare_mats.m
% Load the correct file
% file1 = load('sk_automotive_20221003_164605_correct.skv.mat');

% Load the file in question
% file2 = load('sk_automotive_20221003_164605_out.skv.mat');

% Compare the files using visdiff
% visdiff('data1.mat', 'data2.mat');
% visdiff('sk_automotive_20221003_164605_correct.skv.mat', 'sk_automotive_20221003_164605_out.skv.mat');







% % Create an input parser object
% parser = inputParser;

% % Add arguments to the input parser
% addParameter(parser, '-f1', '', @ischar);
% addParameter(parser, '-f2', '', @ischar);

% % Parse the input arguments
% parse(parser, varargin{:});

% % Get the input file names
% file1 = parser.Results.f1;
% file2 = parser.Results.f2;

% % Check that the input files exist
% if ~exist(file1, 'file')
%     error('Input file %s not found.', file1);
% end

% if ~exist(file2, 'file')
%     error('Input file %s not found.', file2);
% end

% % Compare the files
% visdiff(file1, file2);






function compare_mats(file1, file2)
    % COMPARE_MATS - Script to compare two .mat input files using visdiff and display the results
    %
    % Usage:
    %   $ matlab -r "compare_mats('file1.mat', 'file2.mat');"
    %
    % Inputs:
    %   file1 - Name of the first file to compare
    %   file2 - Name of the second file to compare
    %
    % Examples:
    %   $ matlab -r "compare_mats('sk_automotive_20221003_164605_correct.skv.mat', 'sk_automotive_20221003_164605_out.skv.mat');"
    %   $ matlab -nosplash -r "compare_mats('sk_automotive_20221003_164605_correct.skv.mat', 'sk_automotive_20221003_164605_out.skv.mat');"
    %   $ matlab -nosplash -nodesktop -r "compare_mats('sk_automotive_20221003_164605_correct.skv.mat', 'sk_automotive_20221003_164605_out.skv.mat');"
    %   $ matlab -nodesktop -r "compare_mats('sk_automotive_20221003_164605_correct.skv.mat', 'sk_automotive_20221003_164605_out.skv.mat');"
    
    % % Print the input args
    % fprintf('File 1: %s\n', file1);
    % fprintf('File 2: %s\n', file2);

    % Check that the input files exist
    if ~exist(file1, 'file')
        error('Input file %s not found.', file1);
    end
    
    if ~exist(file2, 'file')
        error('Input file %s not found.', file2);
    end
    
    % Compare the files
    visdiff(file1, file2);
    

File: compare_mats.py
import os
import sys

# Check that two input arguments were provided
if len(sys.argv) != 3:
    print("Usage: python compare_mats.py file1.mat file2.mat")
    sys.exit(1)

# Get the input file names from the command line arguments
file1 = sys.argv[1]
file2 = sys.argv[2]

# Get the absolute paths of the input files
abs_file1 = os.path.abspath(file1)
abs_file2 = os.path.abspath(file2)

# Print the absolute paths for debugging purposes
# print("Absolute path to file 1: %s" % abs_file1)
# print("Absolute path to file 2: %s" % abs_file2)

print(f"Comparing {abs_file1} and {abs_file2}...")

# Check that the input files exist
if not os.path.isfile(abs_file1):
    print("Error: File %s not found" % abs_file1)
    sys.exit(1)

if not os.path.isfile(abs_file2):
    print("Error: File %s not found" % abs_file2)
    sys.exit(1)

# Compare the input files
# matlab -nosplash -nodesktop -wait -r "visdiff('C:\Users\Muhsinun\Desktop\Muhsinun\Repositories\GitHub\facial-bloodflow-tof_live\sk_automotive_20221003_164605_correct.skv.mat', 'C:\Users\Muhsinun\Desktop\Muhsinun\Repositories\GitHub\facial-bloodflow-tof_live\sk_automotive_20221003_164605_out.skv.mat');"
# Run the visdiff command
command = 'matlab -nosplash -nodesktop -wait -r "visdiff(\'%s\', \'%s\');"' % (abs_file1, abs_file2)
os.system(command)